instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/LongRoutesSanityTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.navigation.instrumentation_tests.core
+
+import android.location.Location
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.instrumentation_tests.R
+import com.mapbox.navigation.instrumentation_tests.utils.http.MockDirectionsRequestHandler
+import com.mapbox.navigation.instrumentation_tests.utils.readRawFileText
+import com.mapbox.navigation.instrumentation_tests.utils.withMapboxNavigation
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.getSuccessfulResultOrThrowException
+import com.mapbox.navigation.testing.ui.utils.coroutines.requestRoutes
+import com.mapbox.navigation.testing.ui.utils.coroutines.sdkTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.setNavigationRoutesAsync
+import org.junit.Ignore
+import org.junit.Test
+
+class LongRoutesSanityTest : BaseCoreNoCleanUpTest() {
+
+    override fun setupMockLocation(): Location {
+        return mockLocationUpdatesRule.generateLocationUpdate { }
```

Comment:
It will be [0.0, 0.0]. Which may cause unexpected reroutes at the very beginning. I've run into this problem: if the first point of the route is not the same as the one mocked here, we might still be getting these locations when it's not expected. And you'll have to then add some workarounds to ensure we forgot about this location and only use the valid one. 
I think it doesn't affect this particular test. But I don't think it's good practice. If we add additional checks, we might experience said issues. Let's just use the first coordinate of the route.

Reply:
sounds reasonable, added